src/transport/PeerMessageCounter.h

```diff
@@ -33,6 +33,7 @@ class PeerMessageCounter
 {
 public:
     static constexpr size_t kChallengeSize = 8;
+    static constexpr uint32_t kGroupOffset = static_cast<uint32_t>(1 << 31); // 2^31
```

Comment:
Perhaps 
```suggestion
    static constexpr uint32_t kGroupFutureCounterWindow = static_cast<uint32_t>(1 << 31) - 1; // 2^31 - 1
```
or so?  Makes it clear what this "offset" really means.

Reply:
Offset is used everywhere the spec specify to use 2^31, So I would keep the value as is but add a comment.